Bluetooth: Shell: BR: Add SDP discovery support for AVRCP CT/TG

makeshi opened the pull request:
Add SDP discovery handlers for AVRCP CT/TG roles and corresponding discovery parameters and result handling functions.

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/bredr.c:
```diff
@@ -996,6 +996,92 @@ static uint8_t sdp_a2src_user(struct bt_conn *conn, struct bt_sdp_client_result
+			bt_shell_error("AVRCP CT PSM not found, err %d", err);
+			goto done;
+		}
+		bt_shell_print("AVRCP CT L2CAP PSM param 0x%04x", param);
```
If is the SDP record of CT/TG incorrect?

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/bredr.c:
```diff
@@ -996,6 +996,92 @@ static uint8_t sdp_a2src_user(struct bt_conn *conn, struct bt_sdp_client_result
+
+	conn_addr_str(conn, addr, sizeof(addr));
+
+	if (result && result->resp_buf) {
```
My suggest is decreasing the indents with following reference,
```
	if (result = NULL || result->resp_buf == NULL) {
		bt_shell_print("No SDP AVRCP CT data from remote %s", addr);
		return BT_SDP_DISCOVER_UUID_CONTINUE;
	}

	<process the `result->resp_buf`>
```

makeshi replied:
updated

lylezhu2012 commented on subsys/bluetooth/host/classic/shell/bredr.c:
```diff
@@ -996,6 +996,92 @@ static uint8_t sdp_a2src_user(struct bt_conn *conn, struct bt_sdp_client_result
+
+	conn_addr_str(conn, addr, sizeof(addr));
+
+	if (result && result->resp_buf) {
```
Ditto.

makeshi committed:
Bluetooth: Shell: BR: Add SDP discovery support for AVRCP CT/TG

Add SDP discovery handlers for AVRCP CT/TG roles and corresponding
discovery parameters and a common result handling function.

Signed-off-by: Make Shi <make.shi@nxp.com>

mmahadevan108 commented:
@MarkWangChinese , can you reapprove this PR
